core/src/test/kotlin/com/expediagroup/sdk/domain/rapid/RapidHelpersTest.kt

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2022 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.sdk.domain.rapid
+
+import com.expediagroup.sdk.core.client.BaseRapidClient
+import com.expediagroup.sdk.core.configuration.RapidClientConfiguration
+import io.ktor.client.statement.HttpResponse
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertNull
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+
+class RapidHelpersTest {
+    companion object {
+        private val rapidClient =
+            object : BaseRapidClient(RapidClientConfiguration()) {
+                override suspend fun throwServiceException(
+                    response: HttpResponse,
+                    operationId: String
+                ) {
+                    throw UnsupportedOperationException()
+                }
+            }
+
+        val rapidHelpers = RapidHelpers(rapidClient)
+    }
+
+    @Nested
+    inner class TestExtractToken {
+        @Test
+        fun `Verify extractToken returns token when present`() {
+            assertEquals("12345", rapidHelpers.extractToken("https://www.example.com?token=12345"))
+        }
+
+        @Test
+        fun `Verify extractToken returns null when no token is present`() {
+            assertNull(rapidHelpers.extractToken("https://www.example.com"))
+        }
```

Comment:
Shouldn't we throw an exception if a token is not present? Or if the string is not a url even.

Reply:
I prefer to keep this to the client, adding the exception handling at this level could complicate things. We already added a note that this method could return a null...